indico/web/forms/fields.py

```diff
@@ -476,6 +488,24 @@ def _value(self):
         return self.data or []
 
 
+class ReferencesField(MultipleItemsField):
+    """Extend `MultipleItemsField` to use a list of reference objects as the
+    field value."""
+    def process_formdata(self, valuelist):
+        super(ReferencesField, self).process_formdata(valuelist)
+        if valuelist:
+            self.data = [self.reference_class(reference_type_id=int(r['type']), value=r['value']) for r in self.data]
+
+    def pre_validate(self, form):
+        super(ReferencesField, self).pre_validate(form)
+        for reference in self.data_as_dict:
+            if reference['type'] not in self.choices['type']:
+                raise ValueError(u'Invalid type choice: {}'.format(reference['type']))
```

Comment:
Out of curiosity, Is there a reason why we use `u` instead of `from __future__ import unicode_literals` throughout the whole file?